R/authors.R

```diff
@@ -0,0 +1,103 @@
+#' Create or modify the `Authors@R` field in the DESCRIPTION file
+#'
+#' @description
+#' If the `Authors@R` field is not included in the DESCRIPTION file then `use_author()` adds the field assigns
+#' the author defined by the input parameters. If the `Authors@R` field exists already in the DESCRIPTION then
+#' `use_author()` will add the author defined by the input parameters as an additional new author.
+#'
+#' The `use_author()` function should be used after [create_package()] or [use_description()].
+#'
+#' If you create a lot of packages, consider storing personalized defaults as a
+#' named list in an option named `"usethis.description"`. [use_description()] will automatically
+#' fill using this information. Here's an example of code to include in `.Rprofile`:
+#'
+#' ```
+#' options(
+#'   usethis.description = list(
+#'     `Authors@R` = 'person("Jane", "Doe", email = "jane@example.com", role = c("aut", "cre"),
+#'                           comment = c(ORCID = "YOUR-ORCID-ID"))',
+#'     License = "MIT + file LICENSE",
+#'     Language =  "es"
+#'   )
+#' )
+#' ```
+#'
+#' @param given a character string with the given (first) name of the author.
+#' @param family a character string with the family (last) name of the author.
+#' @param role a character vector specifying the role of the person to be added.
+#' @param email a character string giving an e-mail address of the author to be added.
+#' @param comment a character string providing comments related to the author to be added.
+#'
+#' @export
+#'
+#' @importFrom desc desc_add_author desc_fields desc_get_authors desc_set_authors
+#' @importFrom utils person
+#'
+#' @examples
+#' \dontrun{
+#' # Adds the default Jane Doe to author
+#' use_author()
+#'
+#' # Adds the author information for Ali
+#' use_author(given = "Ali",
+#'           family = "Val",
+#'           role = c("aut", "cre"),
+#'           email = "alval@example.com",
+#'           comment = c(ORCID = "YOUR-ORCID-ID"))
+#'
+#' # Adds a second author Ali2
+#'use_author(given = "Ali2",
+#'           family = "Val2",
+#'           role = c("cph", "fnd"),
+#'           email = "alval2@example.com",
+#'           comment = NULL)
+#' }
+#'
+
+use_author <- function(given = "Jane", family = "Doe", role = c("aut", "cre"), email = "jane@example.com", comment = c(ORCID = "YOUR-ORCID-ID")){
+  # Adapted from use_dependency code and tools provided in the desc package
+  # TODO allow generic authors input like desc_set_authors(authors, file = ".", normalize = FALSE) for multiple author input
+  # TODO create addin to prompt for author information
+  # TODO create a snippet with author information for DESCRIPTION
+  # TODO add tests
+
+  # Set the author as a person
+  author = utils::person(given = given, family = given, role = role, email = email, comment = comment)
+
+  # Obtain the current DESCRIPTION fields
+  current_desc_fields <- desc::desc_fields()
```

Comment:
Are you sure you can't just load up the DESCRIPTION file, grab the current authors (possibly none, but that's going to be very very rare), and just call `desc::desc_add_author()`? What I'm really saying is: does there need to be so much conditional logic?

In general, I'm comfortable assuming that we start with our own output, i.e. that we (usethis) created the DESCRIPTION file. Or that it's not pathological in some way.

Reply:
I tried to incorporate this flow into this version of the function and get rid of the conditional logic. I still have some `if()` to assess what the state of the current description is in order to add an author or error the function.

I now error if the DESCRIPTION has 'Author:' rather than 'Authors@R' and send the user a message. I allow for addition of an author so long as 'Authors@R@' is a field (blank or pre-filled). I do not make changes if the author being input with `use_author()` is already in the DESCRIPTION. I ask the user if they would like to delete the usethis default author (First, Last, ...) if it is found in the DESCRIPTION.

I'd like to eventually change the error from finding 'Author' to prompting the user if they want to replace it with 'Authors@R' and continue adding and also prompt the user to add 'Authors@R' if it is missing and Authors was not found. Not sure if I'm over-thinking this though. I tried to stick with your advice and start by assuming the user used usethis or at least the generic DESCRIPTION generated when using File > New Directory > R Package...